src/RetryMiddleware.php

```diff
@@ -74,14 +74,17 @@ public function __invoke(RequestInterface $request, array $options)
     private function onFulfilled(RequestInterface $req, array $options)
     {
         return function ($value) use ($req, $options) {
-            if (!call_user_func(
+            $result = call_user_func(
                 $this->decider,
                 $options['retries'],
                 $req,
                 $value,
                 null
-            )) {
+            );
+            if (!$result) {
```

Comment:
Why should we do type-casting? As described in phpdoc, decider should return false to reject next retry. Why don't just check $result===false?